app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,29 @@
+class CommentsController < ApplicationController
+
+  def create
+    new_comment = idea_or_comment.comments.new(text: comment_params[:text])
+    new_comment.author = current_user
+    if new_comment.save
+      render partial: 'show', locals: { comment: idea_or_comment }, layout: false if idea_or_comment.is_a? Comment
+      render partial: 'show', locals: { idea: idea_or_comment }, layout: false if idea_or_comment.is_a? Idea
+    else
+      redirect_to idea_path(idea_or_comment) if idea_or_comment.is_a? Idea
+      redirect_to comment_path(idea_or_comment.idea) if idea_or_comment.is_a? Comment
+    end
+  end
+
+  private
+
+  def idea_or_comment
```

Comment:
perhaps if we utilized the term commentable, and then we could change the value of what gets passed based on that commentable type? @rfleury2 what do you think?